### DISCUSSION TITLE
adjusting compression

### DISCUSSION THREAD

**Discussion by sander816** [AzuraCast/AzuraCast#4618] [Q&A]
is there a way to adjust the amount of compression/limiting etc?

**Comment by sander816** [comment]
when i run this command:
root@b6e42635af46:/var/azuracast/stations/#: liquidsoap --list-plugins | grep mb_com
 + ladspa.http:_lsp_plugin_plugins_ladspa_mb_compressor_lr
 + ladspa.http:_lsp_plugin_plugins_ladspa_mb_compressor_mono
 + ladspa.http:_lsp_plugin_plugins_ladspa_mb_compressor_ms
 + ladspa.http:_lsp_plugin_plugins_ladspa_mb_compressor_stereo
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_mb_compressor_lr
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_mb_compressor_mono
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_mb_compressor_ms
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_mb_compressor_stereo

i see there are multiband compressor plugins available, among a lot of other plugins(eq/limit)
 Fast Lookahead limiter by Steve Harris <steve(at)plugin.org.uk>.
 + ladspa.hardlimiter
 + ladspa.http:_lsp_plugin_plugins_ladspa_limiter_mono
 + ladspa.http:_lsp_plugin_plugins_ladspa_limiter_stereo
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_limiter_mono
 + ladspa.http:_lsp_plugin_plugins_ladspa_sc_limiter_stereo
 + ladspa.tap_limiter

is there a way to use these?

**Reply by Vaalyn** [reply]
You can use them via the `Edit Liquidsoap Configuration` page.

See our docs for more information: https://docs.azuracast.com/en/developers/liquidsoap#ladspa-plugins

For reference you can take a look at the mkpascal audio processing script that was adapted to work with AzuraCast (link is also on the docs page but for better visibility I wanted to directly mention it here): https://gist.github.com/SlvrEagle23/43a06ee6624975273fdc903ba4a39998

**Reply by sander816** [reply]
Thats nice, didn't see that it was already working

so i add it in the box below this part?
```
# Normalization and Compression
radio = normalize(target = 0., window = 0.03, gain_min = -16., gain_max = 0., radio)
radio = compress.exponential(radio, mu = 1.0)

radio = fallback(id="safe_fallback", track_sensitive = false, [radio, single(id="error_jingle", "/usr/local/share/icecast/web/error.mp3")])
```

**Reply by Vaalyn** [reply]
Yep, that should be the correct place to put it.

**Comment by sander816** [comment]
https://wiki.daveking.com/index.php/ALSA_LADSPA_Plugin_Chain_For_FM_Transmission

**Comment by radiobellissima** [comment]
Can someone help me I wanted to try the plugin function but if I insert the script the station does not restart. If I type the command I don't see plugins but a long series of parameters after an installation of something.
docker-compose run --rm --user=azuracast stations liquidsoap --list-plugins

Thanks,
Marcello

**Comment by sander816** [comment]
these are not plugins for azuracast. Just a script to load extensions for liquidsoap

**Comment by radiobellissima** [comment]
Yes, I got the term wrong. Azurra says they are already included in the package but I can't get them to work. If I put the script, the radio won't restart anymore. What am I wrong or what did I not understand?
